File: uptmdigital_app/lib/models/nota.dart
class Nota {
  final int idNota;
  final int asignaturaId;
  final int estudianteId;
  final double calificacion;
  final String codigoQR;
  final int? profesorId;

  Nota({
    required this.idNota,
    required this.asignaturaId,
    required this.estudianteId,
    required this.calificacion,
    required this.codigoQR,
    this.profesorId, // Optional, from API
  });

  factory Nota.fromJson(Map<String, dynamic> json) {
    return Nota(
      idNota: json['idNota'] ?? 0,
      asignaturaId: json['asignaturaId'] ?? 0,
      estudianteId: json['estudianteId'] ?? 0,
      calificacion: (json['calificacion'] ?? 0).toDouble(),
      codigoQR: json['codigoQR'] ?? '',
      profesorId: json['profesorId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idNota': idNota,
      'asignaturaId': asignaturaId,
      'estudianteId': estudianteId,
      'calificacion': calificacion,
      'codigoQR': codigoQR,
      'profesorId': profesorId,
    };
  }
}

File: UPTMDigital.API/Models/Nota.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    [Table("Nota")]
    public class Nota
    {
        [Key]
        public int IdNota { get; set; }
        public int AsignaturaId { get; set; }
        public int EstudianteId { get; set; }
        public decimal? Calificacion { get; set; }
        public DateTime? Fecha { get; set; }
        public int? ProfesorId { get; set; }
        public string? CodigoQR { get; set; }

        public Asignatura Asignatura { get; set; } = null!;
        public Estudiante Estudiante { get; set; } = null!;
        public Profesor? Profesor { get; set; }
    }
}